examples/metrics-basic/src/main.rs

```diff
@@ -43,23 +43,21 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     );
 
     // Create a ObservableCounter instrument and register a callback that reports the measurement.
-    let observable_counter = meter
+    let _observable_counter = meter
         .u64_observable_counter("my_observable_counter")
         .with_description("My observable counter example description")
         .with_unit(Unit::new("myunit"))
+        .with_callback(|observer| {
+            observer.observe(
+                100,
```

Comment:
this is still type-safe. i.e passing 100.0 will be blocked by compiler. But the API itself is more consistent with sync versions, where we don't use record_i64 or record_u64 etc. instead they are just record,add.